sdk/core/core-tracing/src/createSpan.ts

```diff
@@ -61,56 +87,40 @@ export function isTracingDisabled(): boolean {
  * const span = createSpan("deleteConfigurationSetting", operationOptions);
  *    // code...
  * span.end();
+ *
+ * // customizing the created span
+ * const span = createSpan("deleteConfigurationSetting", {...operationOptions, spanOptions: { spanKind: SpanKind.SERVER } });
  * ```
  *
  * @hidden
  * @param args - allows configuration of the prefix for each span as well as the az.namespace field.
  */
 export function createSpanFunction(args: CreateSpanFunctionArgs) {
-  return function<T extends { tracingOptions?: OperationTracingOptions }>(
+  // Internally we can provide spanOptions to configure newly created spans, but these spanOptions are not part of our client library public API.
+  return function<
+    T extends { tracingOptions?: OperationTracingOptions } & { spanOptions?: SpanOptions }
+  >(
     operationName: string,
-    operationOptions: T | undefined
-  ): { span: Span; updatedOptions: T } {
-    const tracer = getTracer();
-    const tracingOptions = operationOptions?.tracingOptions || {};
-    const spanOptions: SpanOptions = {
-      kind: SpanKind.INTERNAL,
-      ...tracingOptions.spanOptions
-    };
+    operationOptions?: T
+  ): { span: Span; updatedOptions: Omit<T, "spanOptions"> } {
+    const { tracingOptions, spanOptions, otherOptions } = pluckOptions(operationOptions || {});
 
     const spanName = args.packagePrefix ? `${args.packagePrefix}.${operationName}` : operationName;
-
-    let span: Span;
-    if (isTracingDisabled()) {
-      span = trace.wrapSpanContext(INVALID_SPAN_CONTEXT);
-    } else {
-      span = tracer.startSpan(spanName, spanOptions, tracingOptions.tracingContext);
-    }
+    let context = tracingOptions?.tracingContext || otContext.active();
+    const span = startSpan(spanName, spanOptions, context);
 
     if (args.namespace) {
       span.setAttribute("az.namespace", args.namespace);
-    }
-
-    let newSpanOptions = tracingOptions.spanOptions || {};
-
-    if (span.isRecording() && args.namespace) {
-      newSpanOptions = {
-        ...tracingOptions.spanOptions,
-        attributes: {
-          ...spanOptions.attributes,
-          "az.namespace": args.namespace
-        }
-      };
+      context = context.setValue(Symbol.for("az.namespace"), args.namespace);
```

Comment:
TODO: introduce this symbol as a constant that is exported. core-http's tracingPolicy will use to fetch az.namespace